### PR TITLE
shuffle data from hdf5 datasets

### PR DESCRIPTION
**Pull request by PatWie**
The order of read HDF5 files and the order of the entries of the HDF5 files can be shuffled when setting the flag `shuffle` in the hdf5data layer

### CODE REVIEW

**Comment by shelhamer** [comment]
@jeffdonahue can you review and merge if this looks good to you?

File: src/caffe/proto/caffe.proto [hunk 1]
```diff
@@ -499,6 +499,8 @@ message HDF5DataParameter {
   optional string source = 1;
   // Specify the batch size.
   optional uint32 batch_size = 2;
+  // Specify shuffling the order of training data
```
Comment (jeffdonahue) [inline src/caffe/proto/caffe.proto]: Please clarify this comment to explain that the HDF5 files themselves are shuffled but the order within any given file is fixed.

Comment (wieschoo) [inline src/caffe/proto/caffe.proto]: Everything will be shuffled: hdf5 files and entries in these hdf5 files.

Comment (jeffdonahue) [inline src/caffe/proto/caffe.proto]: Thanks, I see that now, my bad.  I think it should still be clarified though -- it's not actually a full shuffle of the dataset (i.e., some orderings of the dataset are impossible to obtain) unless you only have a single HDF5 file (or each HDF5 file only has a single entry).

**Comment by jeffdonahue** [comment]
Replaced by #2118.

**Pull request closed without merging**